dss/operations/params.py

```diff
@@ -0,0 +1,146 @@
+"""
+Get, set, and unset environment variables in the SSM store and in deployed lambda functions
+"""
+import boto3
+import os
+import typing
+import argparse
+import json
+import logging
+
+from dss.operations import dispatch
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_ssm_lambda_environment():
+    ssm_client = boto3.client("ssm")
+    parms = ssm_client.get_parameter(
+        Name=f"/{os.environ['DSS_PARAMETER_STORE']}/{os.environ['DSS_DEPLOYMENT_STAGE']}/environment"
+    )['Parameter']['Value']
+    return json.loads(parms)
+
+def set_ssm_lambda_environment(parms: dict):
+    ssm_client = boto3.client("ssm")
+    ssm_client.put_parameter(
+        Name=f"/dcp/dss/{os.environ['DSS_DEPLOYMENT_STAGE']}/environment",
+        Value=json.dumps(parms),
+        Type="String",
+        Overwrite=True,
+    )
+
+def get_deployed_lambda_environment(name):
+    lambda_client = boto3.client("lambda")
+    return lambda_client.get_function_configuration(FunctionName=name)['Environment']['Variables']
+
+def set_deployed_lambda_environment(name, env: dict):
+    lambda_client = boto3.client("lambda")
+    lambda_client.update_function_configuration(
+        FunctionName=name,
+        Environment={
+            'Variables': env
+        }
+    )
+
+def get_deployed_lambdas():
+    root, dirs, files = next(os.walk(os.path.join(os.environ['DSS_HOME'], "daemons")))
+    functions = [f"{name}-{os.environ['DSS_DEPLOYMENT_STAGE']}" for name in dirs]
+    functions.append(f"dss-{os.environ['DSS_DEPLOYMENT_STAGE']}")
+    lambda_client = boto3.client("lambda")
+    for name in functions:
+        try:
+            _ = lambda_client.get_function(FunctionName=name)
+            yield name
+        except lambda_client.exceptions.ResourceNotFoundException:
+            print(f"{name} not deployed, or does not deploy a Lambda function")
+
+def get_elasticsearch_endpoint():
+    es_client = boto3.client("es")
```

Comment:
we should upgrade these clients to using `dss.util.aws.clients` see `dss/operations/util.py line 8`

Reply:
I don't follow how `dss/operations/util.py` and the `sts` bit should be used to replace `boto3.client(X)`. I'm not having any success doing `sts.client(X)` or `sts(X)`, and I'm a little confused about what `dss/util/aws/clients.py` and `_boto3_loader.py` (in the same dir) are doing.

Reply:
cleared this up - just do `from dss.util.aws.clients import ssm`, then you can use `ssm` in place of a `boto3.client` object.